GSoC-2025/movement: front-end support for filtering module.md

```diff
@@ -0,0 +1,112 @@
+- **Time Efficiency**: Manual data cleaning consumes research time; real-time filtering previews streamline workflows.  
+- **Reproducibility**: GUI-driven workflows ensure standardized filtering across labs.  
+- **Open Science**: Democratizes advanced tools for underrepresented institutions with limited computational resources.  
+
+---
+
+## Implementation Timeline  
+### 1. **Core Deliverables**  
+- **Napari Widget Suite**:  
+  - Dropdown menus for confidence filters (threshold-based outlier removal).  
+  - Sliders/inputs for Median filter (window size) and SavGol filter (window size, polynomial order).  
+  - Real-time overlay of raw vs. filtered data.  
+- **Testing Framework**:  
+  - Unit tests for filter logic and UI components.  
+  - Performance benchmarks.  
+- **Documentation**:  
+  - Jupyter notebook tutorials with sample datasets.  
+  - Contributor guidelines for extending the widget.  
+- **Export Workflows**: Save/load filter configurations as netCDF.  
+- **Video Tutorial**: 5-minute demo for all the filters, covering how and when to use each. 
+
+### 2. **Extended Deliverables (Aspirational)**  
+- **Filter Chaining**: Sequential application of filters (e.g., confidence → median → SavGol).  
+- **Community Presets**: Curated settings for common use cases (e.g., fruit fly locomotion).  
+- **Hardware Acceleration**: GPU-backed filtering.  
+
+---
+
+## GSoC 2025 Timeline  
+### **Community Bonding Period** *(May 8 – June 1)*  
+| Week | Tasks | Hours | Output |  
+|------|-------|-------|--------|  
+| CB1  | Study napari plugin architecture; draft wireframes with mentors | 7 | Environment setup guide |  
+| CB2  | Profile filter performance; create UML diagrams; setup Sphinx docs | 8 | Technical design doc |  
+| CB3  | Finalize widget CSS theme; draft Jupyter notebook | 5 | PR for docs infrastructure |  
+
+### **Coding Period** *(June 2 – August 24)*  
+
+| Week | Focus Area | Critical Deliverables | Hours |  
+|------|------------|------------------------|-------|  
+| 1    | Widget Framework | Base widget class; confidence threshold UI | 15 |  
+| 2    | Median Filter | Rolling window logic; unit tests | 14 |  
+| 3    | SavGol Filter | Polynomial order control; benchmarks | 16 |  
+| 4    | Preview System | Live parameter tuning; visualization | 14 |  
+| 5    | Error Handling | Tooltip explanations; logging , Filter chaining (extended deliverable) | 18 |  
+| 6    | **Midterm** | Cross-platform tests; user docs v1 | 14 |  
+| 7    | Advanced Features | Filter chaining; netCDF schema design for workflows/data | 16 |  
+| 8    | Export Implementation | UI elements for exporting (buttons, file dialogs)<br>Save/load workflows to netCDF<br>Export filtered data to netCDF;<br>Validation tests | 18 |  
+| 9    | Testing & Docs | Codecov integration;<br>Video tutorial (export demo);<br>Jupyter notebook updates | 15 |  
```

Comment:
no need for codecov integration, codecov works out-of-the-box for all our PRs. You only need to write tests, and we anyway require those for each PR.